src/borg/helpers/time.py

```diff
@@ -109,6 +110,48 @@ def format_timedelta(td):
     return txt
 
 
+def calculate_relative_offset(format_string, from_date, earlier=False):
+    """Calculates offset based on a relative marker. 7d (7 days), 8m (8 months)
+    earlier: whether offset should be calculated to an earlier time.
+    """
+    if from_date is None:
+        from_date = archive_ts_now().date()
+
+    if format_string is not None:
+        day_offset_regex = re.compile(r"(?P<offset>\d+)d")
+        month_offset_regex = re.compile(r"(?P<offset>\d+)m")
```

Comment:
you could use a named group for the unit, like `(?P<unit>[md])")`.

then it is just 1 regex match and the rest if just dispatching based on unit == "m", unit =="d", etc. (safes some code duplication).